### ISSUE TITLE
Version installed becomes zipp-0.0.0 when built from source code with python 3.7.7

### ISSUE THREAD

**Issue by Makesh-Gmak** [jaraco/zipp#53]
When we build from zipp-3.1.0 source code, it is getting installed as zipp-0.0.0. Due to which other modules rely on zipp gets impacted. _( We dont want our build server to pull packages from internet, hence building from source code)_

```
[root@makesh-py3x zipp-3.1.0]# /mypath/python/bin/python3.7 setup.py install
running install
running bdist_egg
running egg_info
writing zipp.egg-info/PKG-INFO
writing dependency_links to zipp.egg-info/dependency_links.txt
writing requirements to zipp.egg-info/requires.txt
writing top-level names to zipp.egg-info/top_level.txt
reading manifest file 'zipp.egg-info/SOURCES.txt'
writing manifest file 'zipp.egg-info/SOURCES.txt'
installing library code to build/bdist.freebsd-8.4-RELEASE-amd64/egg
running install_lib
running build_py
copying zipp.py -> build/lib
creating build/bdist.freebsd-8.4-RELEASE-amd64/egg
copying build/lib/zipp.py -> build/bdist.freebsd-8.4-RELEASE-amd64/egg
byte-compiling build/bdist.freebsd-8.4-RELEASE-amd64/egg/zipp.py to zipp.cpython-37.pyc
creating build/bdist.freebsd-8.4-RELEASE-amd64/egg/EGG-INFO
copying zipp.egg-info/PKG-INFO -> build/bdist.freebsd-8.4-RELEASE-amd64/egg/EGG-INFO
copying zipp.egg-info/SOURCES.txt -> build/bdist.freebsd-8.4-RELEASE-amd64/egg/EGG-INFO
copying zipp.egg-info/dependency_links.txt -> build/bdist.freebsd-8.4-RELEASE-amd64/egg/EGG-INFO
copying zipp.egg-info/requires.txt -> build/bdist.freebsd-8.4-RELEASE-amd64/egg/EGG-INFO
copying zipp.egg-info/top_level.txt -> build/bdist.freebsd-8.4-RELEASE-amd64/egg/EGG-INFO
zip_safe flag not set; analyzing archive contents...
creating 'dist/zipp-0.0.0-py3.7.egg' and adding 'build/bdist.freebsd-8.4-RELEASE-amd64/egg' to it
removing 'build/bdist.freebsd-8.4-RELEASE-amd64/egg' (and everything under it)
Processing zipp-0.0.0-py3.7.egg
Removing /mypath//python/lib/python3.7/site-packages/zipp-0.0.0-py3.7.egg
Copying zipp-0.0.0-py3.7.egg to /mypath/python/lib/python3.7/site-packages
Adding zipp 0.0.0 to easy-install.pth file

Installed /mypath//python/lib/python3.7/site-packages/zipp-0.0.0-py3.7.egg
Processing dependencies for zipp==0.0.0
Finished processing dependencies for zipp==0.0.0
```

pip list gives :
```
zipp                     0.0.0
```
Edit: Observed with zipp-2.1.0 also.

**Comment by dtoubelis** [comment]
I'm having the same issue and it looks suspiciously similar to this one - https://github.com/jaraco/configparser/issues/45

**Comment by dtoubelis** [comment]
Upgrading setuptools to 42.0.2 and setuptools_scm to 3.4.2 solves the problem.

**Comment by jaraco** [comment]
That's right, [zipp declares its build dependencies](https://github.com/jaraco/zipp/blob/02b914eca955b8f1912c519303b0abaca4a4dab2/pyproject.toml#L2), which were updated in 0bf3d43d97d5466fddac708e7ca38ba95281c6e7, first released with v2.0.0. Please comment or re-open if you find that updating those dependencies doesn't satisfy the need.

**Comment by Begasus** [comment]
Thanks to this issue I was able to fix the same issue on Haiku when implementing zipp (new port to our python recipes) :+1:
